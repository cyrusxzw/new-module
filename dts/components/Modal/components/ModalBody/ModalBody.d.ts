import { FC } from 'react';
import type { ModalBodyProps } from './ModalBody.types';
declare const ModalBody: FC<ModalBodyProps>;
export { ModalBody };
