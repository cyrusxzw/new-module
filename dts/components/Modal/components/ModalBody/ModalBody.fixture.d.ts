declare const ModalBodyFixture: {
  children: string;
  copy: {
    close: string;
  };
};
export { ModalBodyFixture };
