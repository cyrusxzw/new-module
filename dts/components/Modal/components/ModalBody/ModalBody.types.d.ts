import { ModalAndModalBodyProps } from '../../Modal.types';
declare type ModalBodyCopy = {
  copy?: {
    close?: string;
  };
};
declare type ModalBodyProps = ModalBodyCopy & ModalAndModalBodyProps;
export type { ModalBodyProps, ModalBodyCopy };
