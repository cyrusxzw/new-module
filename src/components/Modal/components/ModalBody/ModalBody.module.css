@import 'styles/variables/index';
@import 'styles/mixins/index';

.base {
  position: relative;
  overflow: scroll;
  width: 100%;
  max-height: calc(100vh - var(--layout-sm-spacing) * 4);
  padding: var(--layout-sm-margin);
  background-color: var(--color-white);
  color: var(--color-grey-10);
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(15px);
  line-height: 1.6;
  pointer-events: auto;
  transform: translateZ(0);

  * {
    &,
    &::before,
    &::after {
      box-sizing: inherit;
    }
  }

  @media (--viewport-md) {
    max-height: calc(100vh - var(--layout-md-spacing) * 4 - 50px);
    padding: 50px;
  }

  &.isVisible {
    pointer-events: auto;
  }
}

.closeIcon {
  .dark & {
    fill: var(--color-dark-label);
  }

  .light & {
    fill: var(--color-light-label);
  }

  .dark .closeButton:hover & {
    fill: var(--color-secondary);
  }

  .light .closeButton:hover & {
    fill: var(--color-primary);
  }
}

.closeButton {
  position: absolute;
  top: 28px;
  right: 30px;
  width: 30px;
  height: 30px;
  transition: outline-offset 300ms var(--easing-ease-out-cubic);

  @media (--viewport-md) {
    top: 40px;
    right: 42px;
    width: 42px;
    height: 42px;
  }

  .dark &:focus-visible {
    outline: 1px dashed var(--color-dark-copy-alternate);
    outline-offset: 2px;
  }

  .light &:focus-visible {
    outline: 1px dashed var(--color-light-copy-alternate);
    outline-offset: 2px;
  }
}
