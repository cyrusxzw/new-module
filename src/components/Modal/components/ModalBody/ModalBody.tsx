import React from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { Button } from '~/components/Button';
import { Icon } from '~/components/Icon';
import type { ModalBodyType } from './ModalBody.types';
import styles from './ModalBody.module.css';

const ModalBody: ModalBodyType = ({
  children,
  className,
  copy,
  onClose,
  isVisible,
  theme,
}) => {
  const currentTheme = useThemeContext(theme, 'dark');
  const classSet = cx(
    styles.base,
    { [styles.isVisible]: isVisible },
    styles[currentTheme],
    className,
  );

  return (
    <div className={classSet}>
      {children}
      <Button
        className={styles.closeButton}
        dataTestRef="MODAL_CLOSE_BUTTON"
        isInline={true}
        onClick={onClose}
        theme={theme}
        title={copy.close}
      >
        <Icon
          className={styles.closeIcon}
          height={14}
          name="close"
          theme={theme}
          width={14}
        />
      </Button>
    </div>
  );
};

export { ModalBody };
