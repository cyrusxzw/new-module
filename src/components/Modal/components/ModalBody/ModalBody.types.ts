import type { ComponentWithChildren } from '~/types';
import type { ModalAndModalBodyProps } from '../../Modal.types';

type ModalBodyCopy = {
  copy?: {
    close?: string;
  };
};

type ModalBodyProps = ModalBodyCopy & ModalAndModalBodyProps;

type ModalBodyType = ComponentWithChildren<ModalBodyProps>;

export type { ModalBodyCopy, ModalBodyProps, ModalBodyType };
