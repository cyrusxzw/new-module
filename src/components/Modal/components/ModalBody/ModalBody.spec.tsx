import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ModalBody } from './ModalBody';
import { ModalBodyFixture } from './ModalBody.fixture';

describe('<ModalBody />', () => {
  it('should be defined', () => {
    expect(ModalBody).toBeDefined();
  });

  it('should render component correctly and match html snapshot', () => {
    const mockFn = jest.fn();

    const { container } = render(
      <ModalBody copy={ModalBodyFixture.copy} isVisible={true} onClose={mockFn}>
        modal body test
      </ModalBody>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render content and fire the close button on click', () => {
    const handleOnClose = jest.fn();

    render(
      <ModalBody
        copy={{ close: 'close modal body button' }}
        isVisible={true}
        onClose={handleOnClose}
      >
        modal body test
      </ModalBody>,
    );

    expect(screen.getByText(/modal body test/i)).toBeTruthy();

    userEvent.click(screen.getByTitle(/close modal body button/i));

    expect(handleOnClose).toHaveBeenCalledTimes(1);
  });
});
