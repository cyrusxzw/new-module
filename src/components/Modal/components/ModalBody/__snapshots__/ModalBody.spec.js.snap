// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalBody /> should render component correctly and match html snapshot 1`] = `
<div>
  <div
    class="base isVisible dark"
  >
    Body Content
    <button
      class="base dark closeButton"
      data-test-ref="MODAL_CLOSE_BUTTON"
      title="Close Modal"
      type="button"
    >
      <svg
        aria-labelledby="close-1"
        class="base close dark closeIcon"
        data-testid="data-testid-Icon"
        focusable="false"
        height="14"
        role="img"
        style="height: 14px; width: 14px;"
        tabindex="-1"
        viewBox="0 0 50 50"
        width="14"
      >
        <title
          id="close-1"
        >
          close
        </title>
        <g>
          <polygon
            class="close"
            points="50,5 45,0 25,20 5,0 0,5 20,25 0,45 5,50 25,30 45,50 50,45 30,25"
          />
        </g>
      </svg>
    </button>
  </div>
</div>
`;
